crates/node/src/networking/p2p/protocol/mod.rs

```diff
@@ -0,0 +1,186 @@
+use eyre::{eyre, Result};
+
+pub mod internal;
+pub mod v0;
+pub mod v1;
+
+pub fn serialize_handshake(msg: internal::Handshake) -> Result<Vec<u8>> {
+    let msg = v0::Handshake::from(msg);
+    bincode::serialize(&msg).map_err(|e| e.into())
+}
+
+pub fn new_serialize_handshake(protocol_version: u64, msg: internal::Handshake) -> Result<Vec<u8>> {
+    match protocol_version {
+        0 => bincode::serialize(&v0::Handshake::from(msg)),
+        1 => bincode::serialize(&v1::Handshake::from(msg)),
+        ver => return Err(eyre!("unknown protocol version: {ver}")),
+    }
+    .map(|mut bs| {
+        let mut data = Vec::with_capacity(bs.len() + 8);
+        data.append(&mut protocol_version.to_be_bytes().to_vec());
+        data.append(&mut bs);
+        data
+    })
+    .map_err(|e| e.into())
+}
```

Comment:
Once this code is running on all nodes, this version will replace the original `serialize_handshake`.